frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -48,40 +144,215 @@ class Chart extends Component<Props, State> {
+            from,
+        };
+    }
+
+    private displayWeek = () => {
+        if (this.state.source !== 'hourly' && this.lineSeries && this.props.dataHourly) {
+            this.lineSeries.setData(this.props.dataHourly);
+        }
+        this.setState(
+            { display: 'week', source: 'hourly' },
+            () => {
+                if (!this.chart) {
+                    return;
+                }
+                const { utcDate, from, to } = this.getUTCRange();
+                from.setUTCDate(utcDate - 7);
+                this.chart.timeScale().setVisibleRange({
+                    from: from.getTime() / 1000 as UTCTimestamp,
+                    to: to.getTime() / 1000 as UTCTimestamp,
+                });
             }
-            this.chart.timeScale().fitContent();
-            window.addEventListener('resize', this.onResize);
+        );
+    }
+
+    private displayMonth = () => {
+        if (this.state.source !== 'daily' && this.lineSeries && this.props.dataDaily) {
+            this.lineSeries.setData(this.props.dataDaily);
         }
+        this.setState(
+            { display: 'month', source: 'daily' },
+            () => {
+                if (!this.chart) {
+                    return;
+                }
+                const { utcMonth, from, to } = this.getUTCRange();
+                from.setUTCMonth(utcMonth - 1);
+                this.chart.timeScale().setVisibleRange({
+                    from: from.getTime() / 1000 as UTCTimestamp,
+                    to: to.getTime() / 1000 as UTCTimestamp,
+                });
+            }
+        );
     }
 
-    public componentWillUnmount() {
-        window.removeEventListener('resize', this.onResize);
+    private displayYear = () => {
+        if (this.state.source !== 'daily' && this.lineSeries && this.props.dataDaily) {
+            this.lineSeries.setData(this.props.dataDaily);
+        }
+        this.setState(
+            { display: 'year', source: 'daily' },
+            () => {
+                if (!this.chart) {
+                    return;
+                }
+                const { utcYear, from, to } = this.getUTCRange();
+                from.setUTCFullYear(utcYear - 1);
+                this.chart && this.chart.timeScale().setVisibleRange({
+                    from: from.getTime() / 1000 as UTCTimestamp,
+                    to: to.getTime() / 1000 as UTCTimestamp,
+                });
+            }
+        );
+    }
+
+    private displayAll = () => {
+        if (this.state.source !== 'daily' && this.lineSeries && this.props.dataDaily) {
+            this.lineSeries.setData(this.props.dataDaily);
+        }
+        this.setState(
+            { display: 'all', source: 'daily' },
+            () => {
+                if (!this.chart) {
+                    return;
+                }
+                const { to } = this.getUTCRange();
+                this.chart.timeScale().setVisibleRange({
+                    from: '2009-01-03',
```

Comment:
does `fitContent()` not work?

Reply:
changed to `this.chart.timeScale().fitContent();`